hoss/dimension_utilities.py

```diff
@@ -422,22 +559,48 @@ def add_index_range(
 
     """
     variable = varinfo.get_variable(variable_name)
-
     range_strings = []
-
-    for dimension in variable.dimensions:
-        dimension_range = index_ranges.get(dimension)
-
-        if dimension_range is not None and dimension_range[0] <= dimension_range[1]:
-            range_strings.append(f'[{dimension_range[0]}:{dimension_range[1]}]')
+    if variable.dimensions == []:
+        override_dimensions = get_override_dimensions(varinfo, [variable_name])
+        if len(override_dimensions) > 0:
```

Comment:
it is to check the else case where the variable is the same as override or the requested variable is the coordinate variable itself

Reply:
Oh, I see. The point is that you need to determine whether to use the `if` or the `else` on L576.

I think this misunderstanding is probably a good indication that this function is written in a way that is too difficult to follow. There are far too many levels of nesting here. I mentioned it on a different comment, but it would be worth reassessing what really is different between the branches and isolating only those pieces.

Reply:
Changes are in commit [756f7c0](https://github.com/nasa/harmony-opendap-subsetter/pull/15/commits/756f7c05d69fc135b5fc39825f5d7e1dde5daada)